hscontrol/types/config.go

```diff
@@ -46,16 +46,9 @@ type Config struct {
 	Log                            LogConfig
 	DisableUpdateCheck             bool
 
-	DERP DERPConfig
+	Database DatabaseConfig
```

Comment:
nit: Can we implement the method to get the connection string on this struct?
```
func (d DatabaseConfig) GetDBConnectionString() (string, error) {
  // Implements the switch case here and returns the DB string for the
  // configured db type, else an error.
}
```
What do you think?

Reply:
I ended up doing it a bit from the other side, I moved the dbString logic into the openDb method and then I passed the whole DatabaseConfig to the database module as that is where it is consumed. Let me know what you think, it should making passing the settings for your PR quite trivial.

Reply:
Yes, that makes things a lot easier. Thanks. 

I tried doing something similar in the PR https://github.com/juanfont/headscale/pull/1583/files#diff-826f8967b23df1209318fba82f94d8ed3eb12136b6cb2d070ddb76d9a059c336R125 but I guess some parts of it are getting refactored in other PRs.

Reply:
Yep, should not be to bad to rebase on top off. I'll try to get @juanfont to review so you can do yours on top.

Reply:
Sounds good. I'll wait for the merge. Thanks, again.